source/addonStore/network.py

```diff
@@ -109,7 +110,11 @@ def download(
 		f.add_done_callback(self._done)
 
 	def _done(self, downloadAddonFuture: Future[Optional[os.PathLike]]):
-		isCancelled = downloadAddonFuture.cancelled() or downloadAddonFuture not in self._pending
+		isCancelled = (
+			downloadAddonFuture.cancelled()
+			or downloadAddonFuture not in self._pending
+			or self._pending[downloadAddonFuture][0].status != AvailableAddonStatus.DOWNLOADING

```

Comment:
can you explain what the status would be here if the action is cancelled? shouldn't this consider `DOWNLOAD_SUCCESS` too?

Reply:
The reason for determining that the status is not equal to `AvailableAddonStatus.DOWNLOADING` here is that the way to interrupt an ongoing download is to remove `listItemVM` from `progress`.

Reply:
This didn't quite answer my question, let me rephrase:

- what statuses are you expecting here?
- why doesn't this check DOWNLOAD_SUCCESS too?

Reply:
The status is not `AvailableAddonStatus.DOWNLOADING` because there is no corresponding status for canceled downloads.

Because actually canceling a download is just using this function to clean up the code of the downloaded file, the DOWNLOAD_SUCCESS case is out of the scope of this function.